docs/content/FormControl.mdx

```diff
@@ -284,41 +284,150 @@ We encourage using `FormControl` alongside all standalone form components like [
 
 ### With validation
 
+<Note>
+  Refer to the{' '}
+  <a href="https://primer.style/design/ui-patterns/forms#validation">
+    "Validation" section of the form interface guidelines
+  </a>{' '}
+  for more information on validation behavior.
+</Note>
 <Note>Validation messages are not used for an individual checkbox or radio form control.</Note>
 
 ```javascript live noinline
-const ValidationExample = () => {
-  const [value, setValue] = React.useState('mona lisa')
-  const [validationResult, setValidationResult] = React.useState()
-  const doesValueContainSpaces = inputValue => /\s/g.test(inputValue)
-  const handleInputChange = e => {
-    setValue(e.currentTarget.value)
+const getNameErrors = value => {
+  if (!value.replace(/\s/g, '').length) {
+    return ['nameEmpty']
   }
 
-  React.useEffect(() => {
-    if (doesValueContainSpaces(value)) {
-      setValidationResult('noSpaces')
-    } else if (value) {
-      setValidationResult('validName')
+  return []
+}
+
+const getGHHandleErrors = value => {
+  const validationKeys = []
+
+  if (!value.replace(/\s/g, '').length) {
+    validationKeys.push('handleEmpty')
+  }
+
+  if (/\s/g.test(value)) {
+    validationKeys.push('handleHasSpaces')
+  }
+
+  return validationKeys
+}
+
+const ValidationOnSubmitExample = () => {
```

Comment:
Help wanted: this covers the functionality I wanted to showcase, but there's a lot of code to look through. I could use some help simplifying the code for this example.